ddlog fails to build in git-worktree environment

Not an urgent issue, but ddlog fails to build when in a git-worktree(1) environment. The problem we hit is that we use `githash` for inferring the SHA-1 we are currently at but `githash` does not support git-worktree. I've filed [an upstream issue](https://github.com/snoyberg/githash/issues/13) to get this fixed, but opening one here as well to track the problem and fix if I can get to it.

```
[27 of 28] Compiling Language.DifferentialDatalog.Version ( src/Language/DifferentialDatalog/Version.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Language/DifferentialDatalog/Version.o )

differential-datalog-test/src/Language/DifferentialDatalog/Version.hs:38:18: error:
    • Exception when trying to run compile-time code:
        GHECouldn'tReadFile "differential-datalog-test/.git/HEAD" differential-datalog-test/.git/HEAD: openBinaryFile: inappropriate type (Not a directory)
      Code: tGitInfoCwd
    • In the Template Haskell splice $$tGitInfoCwd
      In the first argument of ‘giHash’, namely ‘$$tGitInfoCwd’
      In the expression: giHash $$tGitInfoCwd
   |
38 | gitHash = giHash $$tGitInfoCwd
   |                  ^^^^^^^^^^^^^


--  While building custom Setup.hs for package differential-datalog-0.1.0.0 using:
      ~/.stack/setup-exe-cache/x86_64-linux-tinfo6/Cabal-simple_mPHDZzAJ_2.2.0.1_ghc-8.4.4 --builddir=.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1 build lib:differential-datalog exe:ddlog exe:ovsdb2ddlog --ghc-options " -ddump-hi -ddump-to-file -fdiagnostics-color=always"
    Process exited with code: ExitFailure 1
```

--- a/package.yaml
+++ b/package.yaml
@@ -35,7 +35,7 @@ dependencies:
 - wide-word
 - graphviz
 - text
-- githash
+- githash >= 0.1.3.3
 - process
 - casing
 

--- a/stack.yaml
+++ b/stack.yaml
@@ -41,6 +41,11 @@ packages:
 # (e.g., acme-missiles-0.3)
 # extra-deps: []
 
+extra-deps:
+# We need githash 0.1.3.3 in order to have git-worktree(1) support. It
+# needs to be whitelisted here until we upgrade our resolver.
+- githash-0.1.3.3
+
 # Override default flag values for local packages and extra-deps
 # flags: {}
 

